### PR TITLE
Update Dependencies and bug fixes

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This PR is updating all dependencies bar `readable-stream` as it requires some nodeJS modules not available with Hermes.

Highlights:
* Update to RN 0.71.1
* Update to WDB 0.25.1
* Start using the android back hardware handle directly in JS
* Typescript definition for `AvailableScreens`
* Some bug fixes

#### Ticket Link
Also resolves:
* https://mattermost.atlassian.net/browse/MM-49785 fixes: #6994 
* https://mattermost.atlassian.net/browse/MM-49760
* https://mattermost.atlassian.net/browse/MM-49734 #6983 
* https://mattermost.atlassian.net/browse/MM-49722 (FYI @coltoneshaw as this should be released with 2.1)
* https://mattermost.atlassian.net/browse/MM-49685

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on:
* iPhone 12 Pro, iPhone 14 Pro, iPad 11 Pro
* Android Pixel 6 SDK 33 (Android 13)
* Android emulator with SDK's:
    * 24 (Android 7)
    * 27 (Android 8.1)
    * 28 (Android 9)
    * 29 (Android 10)
    * 30 (Android 11)
    * 31 (Android 12)
    * 33 (Android 13)

#### Release Note
```release-note
* Fixed the keyboard changing on iOS when a code block is started
* Fixed the EMM configuration for VPN timeouts to be in `seconds` instead of `ms`
* Fixed Markdown bold strings
* Fixed Mentions markdown to inherit heading fonts
* Fixed a crash when attempting to download a profile image
```

### CODE REVIEW

File: app/screens/onboarding/index.tsx [hunk 1]
```diff
@@ -111,6 +112,19 @@ const Onboarding = ({
         };
     }, []);
 
+    useEffect(() => {
```
Comment (larkox) [inline app/screens/onboarding/index.tsx:115]: Any reason not to use the hook here?

File: app/screens/server/index.tsx [hunk 2]
```diff
@@ -157,16 +158,36 @@ const Server = ({
...
+
         const navigationEvents = Navigation.events().registerNavigationButtonPressedListener(({buttonId}) => {
             if (closeButtonId && buttonId === closeButtonId) {
-                NetworkManager.invalidateClient(url);
-                dismissModal({componentId});
+                dismiss();
             }
         });
 
+        const backHandler = BackHandler.addEventListener('hardwareBackPress', () => {
```
Comment (larkox) [inline app/screens/server/index.tsx:172]: Any reason not to use the hook?

File: app/screens/server/index.tsx [hunk 3]
```diff
@@ -157,16 +158,36 @@ const Server = ({
     }, [componentId, url, dimensions]);
 
     useEffect(() => {
+        const dismiss = () => {
+            NetworkManager.invalidateClient(url);
+            dismissModal({componentId});
+        };
+
         const navigationEvents = Navigation.events().registerNavigationButtonPressedListener(({buttonId}) => {
```
Comment (larkox) [inline app/screens/server/index.tsx:166]: We could use also the hook for button pressed for this.

File: app/screens/snack_bar/index.tsx [hunk 4]
```diff
@@ -115,7 +117,7 @@ const SnackBar = ({barType, componentId, onAction, sourceScreen}: SnackBarProps)
                     width: (SNACK_BAR_WIDTH / 100) * diffWidth,
                 };
                 break;
-            case [Screens.PERMALINK, Screens.MENTIONS, Screens.SAVED_MESSAGES].includes(sourceScreen):
+            case ([Screens.PERMALINK, Screens.MENTIONS, Screens.SAVED_MESSAGES] as AvailableScreens[]).includes(sourceScreen):
```
Comment (larkox) [inline app/screens/snack_bar/index.tsx:120]: 0/5 this line can be less cluttered if we define this (maybe outside of the component) in a constant.
i.e.
```
const caseScreens: AvailableScreens[] = [Screens.PERMALINK, Screens.MENTIONS, Screens.SAVED_MESSAGES];
...
case caseScreens.includes(sourceScreen):
```
(of course, the name of the constant relevant to this context)

File: app/screens/user_profile/user_profile.tsx [hunk 5]
```diff
@@ -56,7 +57,7 @@ const UserProfile = ({
     const {formatMessage, locale} = useIntl();
     const serverUrl = useServerUrl();
     const {bottom} = useSafeAreaInsets();
-    const channelContext = [Screens.CHANNEL, Screens.THREAD].includes(location);
+    const channelContext = ([Screens.CHANNEL, Screens.THREAD] as AvailableScreens[]).includes(location);
```
Comment (larkox) [inline app/screens/user_profile/user_profile.tsx:60]: Same proposal of moving this to a constant. 0/5

File: app/utils/deep_link/index.ts [hunk 6]
```diff
@@ -88,21 +89,22 @@ export async function handleDeepLink(deepLinkUrl: string, intlShape?: IntlShape,
             }
             case DeepLink.Permalink: {
                 const deepLinkData = parsed.data as DeepLinkPermalink;
-                if (NavigationStore.hasModalsOpened() || ![Screens.HOME, Screens.CHANNEL, Screens.GLOBAL_THREADS, Screens.THREAD].includes(NavigationStore.getVisibleScreen())) {
+                if (NavigationStore.hasModalsOpened() ||
+                !([Screens.HOME, Screens.CHANNEL, Screens.GLOBAL_THREADS, Screens.THREAD] as AvailableScreens[]).includes(NavigationStore.getVisibleScreen())) {
```
Comment (larkox) [inline app/utils/deep_link/index.ts:93]: Same proposal of moving this to a constant.

File: app/utils/deep_link/index.ts [hunk 7]
```diff
@@ -88,21 +89,22 @@ export async function handleDeepLink(deepLinkUrl: string, intlShape?: IntlShape,
             }
             case DeepLink.Permalink: {
                 const deepLinkData = parsed.data as DeepLinkPermalink;
-                if (NavigationStore.hasModalsOpened() || ![Screens.HOME, Screens.CHANNEL, Screens.GLOBAL_THREADS, Screens.THREAD].includes(NavigationStore.getVisibleScreen())) {
+                if (NavigationStore.hasModalsOpened() ||
```
Comment (larkox) [inline app/utils/deep_link/index.ts:92]: 0/5 add a few line breaks so this reads as
```
if (
    Condition1 ||
    Condition2
) {
```
instead of
```
if (Condition 1 ||
Condition2) {
```

File: app/utils/deep_link/index.ts [hunk 8]
```diff
@@ -88,21 +89,22 @@ export async function handleDeepLink(deepLinkUrl: string, intlShape?: IntlShape,
...
                     await dismissAllModalsAndPopToRoot();
                 }
                 showPermalink(existingServerUrl, deepLinkData.teamName, deepLinkData.postId);
                 break;
             }
             case DeepLink.Plugin: {
-                const deepLinkData = parsed.data as DeepLinkPlugin;
-                showModal('PluginInternal', deepLinkData.id, {link: location});
+                // const deepLinkData = parsed.data as DeepLinkPlugin;
+                // showModal('PluginInternal', deepLinkData.id, {link: location});
```
Comment (larkox) [inline app/utils/deep_link/index.ts:101]: Any reason to leave this commented? We should either remove it, or create a todo with ticket associated.

File: app/utils/file/index.ts [hunk 9]
```diff
@@ -477,52 +477,52 @@ export const fileExists = async (path: string) => {
...
-    switch (hasPermissionToStorage) {
-        case Permissions.RESULTS.DENIED:
-            permissionRequest = await Permissions.request(storagePermission);
-            return permissionRequest === Permissions.RESULTS.GRANTED;
-        case Permissions.RESULTS.BLOCKED: {
-            const applicationName = DeviceInfo.getApplicationName();
-            const title = intl.formatMessage(
-                {
-                    id: 'mobile.storage_permission_denied_title',
+    if (Platform.OS === 'android' && Platform.Version < 33) {
```
Comment (larkox) [inline app/utils/file/index.ts:480]: Is this the same function as `hasStoragePermission` with just changing the version, the permission itself, and the alert message?

I wonder if we should unify them.

Also... sanity check... storage check is on version 32, this one is on version 33. Is that correct?

File: app/utils/gallery/index.ts [hunk 10]
```diff
@@ -105,10 +106,10 @@ export function measureItem(ref: React.RefObject<any>, sharedValues: GalleryMana
...
         const measurements = measure(ref);
 
-        sharedValues.x.value = measurements.pageX;
-        sharedValues.y.value = measurements.pageY;
-        sharedValues.width.value = measurements.width;
-        sharedValues.height.value = measurements.height;
+        sharedValues.x.value = measurements?.pageX || 999999;
+        sharedValues.y.value = measurements?.pageY || 999999;
+        sharedValues.width.value = measurements?.width || 0;
+        sharedValues.height.value = measurements?.height || 0;
```
Comment (larkox) [inline app/utils/gallery/index.ts:112]: Curious about what these default values try to accomplish

File: app/utils/navigation/index.ts [hunk 11]
```diff
@@ -1,14 +1,16 @@
...
 
 import {Screens, ServerErrors} from '@constants';
 import {isServerError} from '@utils/errors';
 
-export const appearanceControlledScreens = new Set([
+import type {AvailableScreens} from '@typings/screens/navigation';
+import type {IntlShape} from 'react-intl';
+
+export const appearanceControlledScreens = new Set<AvailableScreens>([
     Screens.ONBOARDING,
```
Comment (larkox) [inline app/utils/navigation/index.ts:14]: Probably not for this PR (already big enough) but now that we have the `AvailableScreens` type, I wonder if we should stop using the `Screens` constant and just rely on type checks. 0/5.

File: types/database/database.ts [hunk 12]
```diff
@@ -59,7 +58,7 @@ export type OperationArgs = {
   transformer: (args: TransformerArgs) => Promise<Model>;
 };
 
-export type Models = Array<Class<Model>>;
+export type Models = Model[];
```
Comment (larkox) [inline types/database/database.ts:61]: Why this change?

**Review by larkox — COMMENTED** [review]
In general looks great! Holding back the approval just because of the number of comments.

DISCLAIMER: I trust you on the Fastlane file and the patches. If you want me to review them thorough let me know.

File: android/app/src/debug/java/com/mattermost/flipper/ReactNativeFlipper.java [hunk 13]
```diff
@@ -4,7 +4,7 @@
  * <p>This source code is licensed under the MIT license found in the LICENSE file in the root
  * directory of this source tree.
  */
-package com.rn;
+package com.mattermost.flipper;
```
[the remark this answers is not in the document]
Comment (enahum) [reply android/app/src/debug/java/com/mattermost/flipper/ReactNativeFlipper.java:7]: Is the name of the package and is following the RN template

File: android/app/src/main/java/com/mattermost/share/ShareUtils.java [hunk 14]
```diff
@@ -90,22 +90,26 @@ private static void addVideoThumbnailToMap(File cacheDir, Context context, Writa
...
             }
 
-            retriever.setDataSource(decodedPath.replace("file://", ""));
-        } else if (filePath.contains("content://")) {
-            retriever.setDataSource(context, Uri.parse(filePath));
+            Bitmap image = retriever.getFrameAtTime(time * 1000, MediaMetadataRetriever.OPTION_CLOSEST_SYNC);
+            retriever.release();
+            return image;
+        } catch (Exception e) {
+            throw new IllegalStateException("File doesn't exist or not supported");
```
[the remark this answers is not in the document]
Comment (enahum) [reply android/app/src/main/java/com/mattermost/share/ShareUtils.java:112]: Forced to add it by the kotlin compiler

File: app/actions/remote/command.ts [hunk 15]
```diff
@@ -18,6 +16,9 @@ import {showAppForm} from '@screens/navigation';
 import {handleDeepLink, matchDeepLink} from '@utils/deep_link';
 import {tryOpenURL} from '@utils/url';
 
+import type {Client} from '@client/rest';
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/actions/remote/command.ts:19]: Yes, new rule included in the RN template

File: app/components/autocomplete/autocomplete.tsx [hunk 16]
```diff
@@ -31,8 +31,9 @@ const getStyleFromTheme = makeStyleSheetFromTheme((theme) => {
             elevation: 3,
         },
         shadow: {
-            shadowColor: '#000',
-            shadowOpacity: 0.12,
+            backgroundColor: theme.centerChannelBg,
+            shadowColor: 'red',
+            shadowOpacity: 1,
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/components/autocomplete/autocomplete.tsx:36]: Oops, my bad

File: app/components/post_draft/post_input/post_input.tsx [hunk 17]
```diff
@@ -307,24 +296,24 @@ export default function PostInput({
...
+            onPaste={onPaste}
             onSelectionChange={handlePostDraftSelectionChanged}
             placeholder={intl.formatMessage(getPlaceHolder(rootId), {channelDisplayName})}
             placeholderTextColor={changeOpacity(theme.centerChannelColor, 0.5)}
-            multiline={true}
-            onBlur={onBlur}
-            onFocus={onFocus}
-            blurOnSubmit={false}
+            ref={inputRef}
+            smartPunctuation='disable'
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/components/post_draft/post_input/post_input.tsx:311]: In our case always, it also affects slash commands with flags etc..

File: app/components/syntax_highlight/renderer.tsx [hunk 18]
```diff
@@ -137,6 +137,8 @@ const CodeHighlightRenderer = ({defaultColor, digits, fontFamily, fontSize, rows
             <FlatList
                 data={rows}
                 renderItem={renderItem}
+
+                //@ts-expect-error key not defined in types
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/components/syntax_highlight/renderer.tsx:141]: This should render once, but I'll change it to create it once anyway

File: app/database/operator/server_data_operator/handlers/posts_in_channel.ts [hunk 19]
```diff
@@ -42,7 +43,7 @@ const PostsInChannelHandler = (superclass: any) => class extends superclass {
         for (const chunk of existingChunks) {
             if (newChunk.earliest <= chunk.earliest && newChunk.latest >= chunk.latest) {
                 if (!prepareRecordsOnly) {
-                    newChunk.prepareUpdate();
+                    newChunk.prepareUpdate(emptyFunction);
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/database/operator/server_data_operator/handlers/posts_in_channel.ts:46]: Same as https://github.com/Nozbe/WatermelonDB/pull/1318

File: app/queries/servers/post.ts [hunk 20]
```diff
@@ -20,24 +20,17 @@ const {SERVER: {POST, POSTS_IN_CHANNEL, POSTS_IN_THREAD, PREFERENCE}} = MM_TABLE
 
 export const prepareDeletePost = async (post: PostModel): Promise<Model[]> => {
     const preparedModels: Model[] = [post.prepareDestroyPermanently()];
-    const relations: Array<Query<Model>> = [post.drafts, post.postsInThread];
-    for await (const relation of relations) {
+    const relations: Array<Query<Model>> = [post.drafts, post.postsInThread, post.files, post.reactions];
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/queries/servers/post.ts:23]: Magic... the Query class has an implicit `then` so when you use await it returns the model instead of the query.. I spent hours trying to understand why TS was yelling at me

File: app/queries/servers/thread.ts [hunk 21]
```diff
@@ -193,7 +193,7 @@ export const queryThreads = (database: Database, teamId?: string, onlyUnreads =
     ];
 
     // Only get threads from available channel
-    const channelCondition: Q.Condition[] = [
+    const channelCondition: Q.Where[] = [
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/queries/servers/thread.ts:196]: I see no reason to remove it at this point

File: app/screens/edit_post/edit_post_input/index.tsx [hunk 22]
```diff
@@ -29,23 +32,24 @@ export type EditPostInputRef = {
...
-    keyboardType, message, onChangeText, onTextSelectionChange, hasError,
+    message, onChangeText, onTextSelectionChange, hasError,
 }: PostInputProps, ref) => {
     const intl = useIntl();
     const theme = useTheme();
     const styles = getStyleSheet(theme);
     const {height} = useWindowDimensions();
+    const managedConfig = useManagedConfig<ManagedConfig>();
     const textInputHeight = (height / 2) - HEIGHT_DIFF;
+    const disableCopyAndPaste = managedConfig.copyAndPasteProtection === 'true';
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/screens/edit_post/edit_post_input/index.tsx:50]: Exactly

File: app/screens/edit_post/edit_post_input/index.tsx [hunk 23]
```diff
@@ -68,17 +72,20 @@ const EditPostInput = forwardRef<EditPostInputRef, PostInputProps>(({
...
                 disableFullscreenUI={true}
                 keyboardAppearance={getKeyboardAppearanceFromTheme(theme)}
-                keyboardType={keyboardType}
                 multiline={true}
                 onChangeText={onChangeText}
+                onPaste={emptyFunction}
                 onSelectionChange={onSelectionChange}
                 placeholder={intl.formatMessage({id: 'edit_post.editPost', defaultMessage: 'Edit the post...'})}
-                placeholderTextColor={changeOpacity(theme.centerChannelColor, 0.4)}
+                placeholderTextColor={changeOpacity(theme.centerChannelColor, 0.5)}
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/screens/edit_post/edit_post_input/index.tsx:85]: Yes, it matches the other ones

File: app/screens/edit_post/edit_post_input/index.tsx [hunk 24]
```diff
@@ -68,17 +72,20 @@ const EditPostInput = forwardRef<EditPostInputRef, PostInputProps>(({
...
+                ref={inputRef}
+                smartPunctuation='disable'
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/screens/edit_post/edit_post_input/index.tsx:87]: Always

File: app/screens/edit_server/index.tsx [hunk 25]
```diff
@@ -93,6 +89,9 @@ const EditServer = ({closeButtonId, componentId, server, theme}: ServerProps) =>
         setDisplayNameError(undefined);
     }, []);
 
+    useNavButtonPressed(closeButtonId || '', componentId, close, []);
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/screens/edit_server/index.tsx:92]: I rather jeep it like this so that the coder is setting the value on purpose

File: app/screens/forgot_password/index.tsx [hunk 26]
```diff
@@ -268,6 +269,8 @@ const ForgotPassword = ({serverUrl, theme}: Props) => {
         return () => unsubscribe.remove();
     }, [dimensions]);
 
+    useAndroidHardwareBackHandler(Screens.FORGOT_PASSWORD, onReturn);
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/screens/forgot_password/index.tsx:272]: We could, although in this case, this is the only instance used, so 0/5

File: app/screens/gallery/index.tsx [hunk 27]
```diff
@@ -63,6 +64,8 @@ const GalleryScreen = ({galleryIdentifier, hideActions, initialIndex, items}: Pr
         setLocalIndex(index);
     }, []);
 
+    useAndroidHardwareBackHandler(Screens.GALLERY, close);
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/screens/gallery/index.tsx:67]: Not really needed, unless we plan to use id's instead of names

File: app/screens/home/channel_list/categories_list/categories/categories.test.tsx [hunk 28]
```diff
@@ -17,11 +19,15 @@ describe('components/channel_list/categories', () => {
     });
 
     it('render without error', () => {
+        jest.useFakeTimers();
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/screens/home/channel_list/categories_list/categories/categories.test.tsx:22]: Basically we had some silent errors complaining about the setTimeout changing state when the component mounts, this fixes the silent errors

File: app/screens/home/search/search.tsx [hunk 29]
```diff
@@ -232,7 +232,7 @@ const SearchScreen = ({teamId, teams}: Props) => {
             paddingTop: scrollPaddingTop,
             flexGrow: 1,
             justifyContent: (resultsLoading || loading) ? 'center' : 'flex-start',
-        } as ViewProps;
+        } as ViewStyle;
     }, [loading, resultsLoading, scrollPaddingTop]);
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/screens/home/search/search.tsx:236]: Not sure what you are saying here

File: app/screens/mfa/index.tsx [hunk 30]
```diff
@@ -181,6 +182,10 @@ const MFA = ({config, goToHome, license, loginId, password, serverDisplayName, s
         return () => unsubscribe.remove();
     }, [dimensions]);
 
+    useAndroidHardwareBackHandler(Screens.MFA, () => {
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/screens/mfa/index.tsx:185]: Same answer :)

File: app/screens/onboarding/index.tsx (hunk 1 above)
Comment (enahum) [reply app/screens/onboarding/index.tsx:115]: Yes, the hook execute a callback and returns true as it was handled, in this case we want to also return false in the event it wasn't

File: app/screens/server/index.tsx (hunk 2 above)
Comment (enahum) [reply app/screens/server/index.tsx:172]: Same as the other one

File: app/utils/deep_link/index.ts (hunk 8 above)
Comment (enahum) [reply app/utils/deep_link/index.ts:101]: I'm not sure who owns this but the screen does not exist, so I commented the lines to remove the ts errors

File: app/utils/file/index.ts (hunk 9 above)
Comment (enahum) [reply app/utils/file/index.ts:480]: Is the same, although not needed for SDK 33 as that permission no longer exists

File: app/utils/gallery/index.ts (hunk 10 above)
Comment (enahum) [reply app/utils/gallery/index.ts:112]: Set some values when the item could not be measured as is outside of the viewport

File: types/database/database.ts (hunk 12 above)
Comment (enahum) [reply types/database/database.ts:61]: https://github.com/Nozbe/WatermelonDB/pull/1318 removed Class

File: app/queries/servers/post.ts (hunk 20 above)
[the remark this answers is not in the document]
Comment (larkox) [reply app/queries/servers/post.ts:23]: And what if it crashes because the query doesn't work? (missing ids or whatever)
Or that cannot happen here?

File: app/screens/home/search/search.tsx (hunk 29 above)
[the remark this answers is not in the document]
Comment (larkox) [reply app/screens/home/search/search.tsx:236]: Instead of using as, typing the variable the useMemo has to return, so we don't need the `as`.

File: app/queries/servers/post.ts (hunk 20 above)
[the remark this answers is not in the document]
Comment (enahum) [reply app/queries/servers/post.ts:23]: is a query that executes `fetch` so worse case scenario you get an empty array

File: app/utils/deep_link/index.ts (hunk 8 above)
Comment (larkox) [reply app/utils/deep_link/index.ts:105]: That should be integrations, but not sure this was ever properly implemented. Either way... if the screen does not exist, there may be a regression in stuff like connecting to plugins that require oauth (I think this screen was created in v1 for that purpose).

Comment (enahum) [reply app/utils/deep_link/index.ts:105]: @javaguirre can you create a ticket for this?

Comment (javaguirre) [reply app/utils/deep_link/index.ts:105]: https://mattermost.atlassian.net/browse/MM-49846

**Review by larkox — APPROVED** [review]
LGTM!

File: app/utils/deep_link/index.ts [hunk 31]
```diff
@@ -88,21 +91,24 @@ export async function handleDeepLink(deepLinkUrl: string, intlShape?: IntlShape,
...
+                ) {
                     await dismissAllModalsAndPopToRoot();
                 }
                 showPermalink(existingServerUrl, deepLinkData.teamName, deepLinkData.postId);
                 break;
             }
             case DeepLink.Plugin: {
-                const deepLinkData = parsed.data as DeepLinkPlugin;
-                showModal('PluginInternal', deepLinkData.id, {link: location});
+                // const deepLinkData = parsed.data as DeepLinkPlugin;
```
Comment (avinashlng1080) [inline app/utils/deep_link/index.ts:104]:
```suggestion
                // https://mattermost.atlassian.net/browse/MM-49846
                // const deepLinkData = parsed.data as DeepLinkPlugin;
```

**Review by avinashlng1080 — APPROVED** [review]
LGTM. Just a few comments here and there.

**Comment by avinashlng1080** [comment]
@jfrerich Please rebase with this PR as it modifies the UserProfile component.

**Comment by avinashlng1080** [comment]
Merge conflicts on this PR

**Comment by enahum** [comment]
@cwarnermm what docs?

File: app/utils/markdown/index.ts [hunk 32]
```diff
@@ -213,30 +214,6 @@ export function escapeRegex(text: string) {
     return text.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&');
 }
 
-export function switchKeyboardForCodeBlocks(value: string, cursorPosition: number) {
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/utils/markdown/index.ts:216]: Yes, we don't need it, as now we are simply disabling smart punctuation on iOS keyboard

File: android/app/build.gradle [hunk 33]
```diff
@@ -95,33 +63,35 @@ if (System.getenv("SENTRY_ENABLED") == "true") {
...
- * This should be set on project.ext.react and that value will be read here. If it is not set
- * on project.ext.react, JavaScript will not be compiled to Hermes Bytecode
- * and the benefits of using Hermes will therefore be sharply reduced.
+ * The international variant includes ICU i18n library and necessary data
+ * allowing to use e.g. `Date.toLocaleString` and `String.localeCompare` that
+ * give correct results when using with locales other than en-US. Note that
+ * this variant is about 6MiB larger per architecture than default.
  */
-def enableHermes = project.ext.react.get("enableHermes", false);
+def jscFlavor = 'org.webkit:android-jsc-intl:+'
```
[the remark this answers is not in the document]
Comment (enahum) [reply android/app/build.gradle:89]: Ok two things:
1. This app is being built with Hermes not JSC
2. In case someone ones to compile the app using JSC they still need the international variant as it contains the intl functions needed to apply the translations like `Date.toLocaleString` and `String.localeCompare` that you'll find all over the codebase, without it, the app will simply crash with an `undefined` function.

File: app/components/post_draft/post_input/post_input.tsx [hunk 34]
```diff
@@ -180,11 +180,6 @@ export default function PostInput({
     const handlePostDraftSelectionChanged = useCallback((event: NativeSyntheticEvent<TextInputSelectionChangeEventData> | null, fromHandleTextChange = false) => {
         const cp = fromHandleTextChange ? cursorPosition : event!.nativeEvent.selection.end;
 
-        if (Platform.OS === 'ios') {
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/components/post_draft/post_input/post_input.tsx:183]: We used to detect if the caret (cursor) position was inside a code block and when that was true, we changed the keyboard to use the `email` keyboard type as that keyboard did not convert double dash to an emdash as the regular keyboard does.

We added a property to [react-native-paste-input](https://github.com/mattermost/react-native-paste-input/commit/b65a7d7779eacf38ac51297e4352262e0f48c639) to disable smart punctuation so that now double dash remains as `--` thus there is no need to change the keyboard type anymore fixing https://mattermost.atlassian.net/browse/MM-49685

File: app/database/manager/__mocks__/index.ts [hunk 35]
```diff
@@ -45,14 +45,14 @@ class DatabaseManager {
     private readonly serverModels: Models;
 
     constructor() {
-        this.appModels = [InfoModel, GlobalModel, ServersModel];
+        this.appModels = [InfoModel, GlobalModel, ServersModel] as unknown[] as Model[];
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/database/manager/__mocks__/index.ts:48]: yes it does..

**Comment by cwarnermm** [comment]
@enahum - I flagged this one as needing docs because of the RN dependency update, thinking that I'd seen versions specified in the docs. After a quick sweep, I don't see such references and will remove the docs/needed label :)

**Comment by enahum** [comment]
Manual cherry-pick to 2.0 done
